test/csuite/wt11126_compile_config/main.c

```diff
@@ -316,39 +350,60 @@ int
 main(int argc, char *argv[])
 {
     TEST_OPTS *opts, _opts;
-    uint64_t base_ns, ns, nsecs[N_VARIANTS];
-    int variant, runs;
+    uint64_t base_ns, ns, *nsecs, total_nsecs;
+    uint32_t i, nthreads, variant;
     const char *compiled_config, **compiled_config_array;
+    THREAD_OPTS *thread_opts;
+    pthread_t *tids;
 
     opts = &_opts;
     memset(opts, 0, sizeof(*opts));
     testutil_check(testutil_parse_opts(argc, argv, opts));
+    nthreads = (u_int)opts->nthreads;
+    if (nthreads == 0)
+        nthreads = 1; /* default value */
+
     testutil_recreate_dir(opts->home);
     testutil_check(wiredtiger_open(opts->home, NULL,
       "create,statistics=(all),statistics_log=(json,on_close,wait=1)", &opts->conn));
-    testutil_check(opts->conn->open_session(opts->conn, NULL, NULL, &opts->session));
 
     begin_transaction_medium_init();
     begin_transaction_fast_init(opts->conn, &compiled_config);
     begin_transaction_fast_alternate_init(opts->conn, &compiled_config_array);
 
-    memset(nsecs, 0, sizeof(nsecs));
+    nsecs = dcalloc(N_VARIANTS * nthreads, sizeof(*nsecs));
+    thread_opts = dcalloc(nthreads, sizeof(*thread_opts));
+    tids = dcalloc(nthreads, sizeof(*tids));
 
-    /* Run the test, alternating the variants of tests. */
-    for (runs = 0; runs < N_RUNS; ++runs)
-        for (variant = 0; variant < N_VARIANTS; ++variant) {
-            do_config_run(
-              opts, variant, compiled_config, compiled_config_array, runs == 0, &nsecs[variant]);
-        }
+    for (i = 0; i < nthreads; ++i) {
+        thread_opts[i].test_opts = opts;
+        thread_opts[i].compiled_config = compiled_config;
+        thread_opts[i].compiled_config_array = compiled_config_array;
+        thread_opts[i].nsecs = &nsecs[i * N_VARIANTS];
+        pthread_create(&tids[i], NULL, benchmark_thread, &thread_opts[i]);
+    }
+    for (i = 0; i < nthreads; ++i)
+        pthread_join(tids[i], NULL);
 
     printf("number of calls: %d\n", N_CALLS * N_RUNS);
-    base_ns = ns = nsecs[0] / (N_CALLS * N_RUNS);
+    printf("number of threads: %" PRIu32 "\n", nthreads);
```

Comment:
Should this be above the create/join to give output before the benchmark starts?